src/networking.c

```diff
@@ -959,9 +958,20 @@ void copyClientOutputBuffer(client *dst, client *src) {
     listRelease(dst->reply);
     dst->sentlen = 0;
     dst->reply = listDup(src->reply);
-    memcpy(dst->buf,src->buf,src->bufpos);
-    dst->bufpos = src->bufpos;
     dst->reply_bytes = src->reply_bytes;
+    if ((long)dst->buf_usable_size >= src->bufpos) {
+        memcpy(dst->buf,src->buf,src->bufpos);
+        dst->bufpos = src->bufpos;
+    } else {
+        /* We can't use buf of client, so create a new clientReplyBlock. */
```

Comment:
as @oranagra said, in change 2, in order to test, in my first commit, i change condition to test this code. i remember we shouldn't assume the real alloc size of `zmalloc`, so it may happen. Just care about it is very small possibility, so this implementation is not efficient.

Reply:
following up on this topic in this thread (responding to later comments)
if we want, we can try to resolve that concern and have an efficient code in both scenarios.
if the `buf` is completely empty (and maybe also the new payload to add is bigger than `PROTO_REPLY_CHUNK_BYTES`), we create a new node, and leave `buf` completely unused. but if `buf` is not empty (it'll have a `write` call anyway), we can utilize it's space.

i'm not certain this complication is worth it, although that's probably just 2 lines of code, and another 2 lines of comment.